openmdao/core/driver.py

```diff
@@ -804,3 +836,63 @@ def _setup_simul_coloring(self, mode='fwd'):
                     del dvdict[dv]
                     dv = prom2abs[dv][0]
                     dvdict[dv] = col_dict
+
+    def _pre_run_model_debug_print(self):
+        """
+        Set up metadata for simultaneous derivative solution.
```

Comment:
I don't think this docstring (or the other one) is right

Reply:
Also, it occurs to me that you don't really need to record desvars before the model runs. They don't change. You can put all of this code into the _post_rn_model_debug_print, and get rid of this function.